libr/magic/funcs.c

```diff
@@ -322,7 +322,7 @@ const char *__magic_file_getbuffer(RMagic *ms) {
 
 int __magic_file_check_mem(RMagic *ms, unsigned int level) {
 	if (level >= ms->c.len) {
-		size_t len = (ms->c.len += 20) * sizeof (*ms->c.li);
+		size_t len = (ms->c.len = 20 + level) * sizeof (*ms->c.li);
```

Comment:
maybe better like this?

```suggestion
		ms->c.len += 20 + level;
		size_t len = ms->c.len * sizeof (*ms->c.li);
```

Reply:
Not so sure tbh. I'm running a clone detection tool and found the clone then suggested the fix based on the original fix. And the original fix is '=' not '+='. Anw, you should have a better grasp than I do about this.

Reply:
ah yes, the level must be larger than c.len, there's no need to make the increment exponential. i was just splitting the line in two for readability. the code quality of libfile is pretty bad, i would like to rewrite it completely at some point, but at least having a working implementation was good. i was not planning to keep in sync